src/main.cc

```diff
@@ -393,6 +394,7 @@ int main(int argc, char** argv, char** envp)
                     }
 
                     ///  \todo fix this for SIGHUP
+                    auto config = std::static_pointer_cast<Config>(configManager);
```

Comment:
This should not be needed: `shared_ptr` can be constructed from a derived class so `make_shared<Server>(configManager)` should work.

~I guess you ran into issues because `Server` accepts a reference to `shared_ptr` which is not the same thing. I would change `Server` constructor to take `shared_ptr` by value and then move into a member. It will also do one less copy,..~

Apparently it does accept by value, so I see no reason why it won't work....